connection.go

```diff
@@ -101,9 +101,14 @@ func (p *Pool) Exec(req *Request) ([]byte, error) {
 	conn := p.Get()
 	defer p.Put(conn)
 start:
-	if err := websocket.Message.Send(conn.ws, requestMessage); err != nil {
-		p.logger.Log("sendMessageErr", err)
-		return nil, err
+	err = websocket.Message.Send(conn.ws, requestMessage)
+	if err != nil {
+		conn.ws, err = p.createSocket()
+		if err != nil {
+			return nil, err
+		}
+		p.logger.Log("socketRecovered", conn.id)
+		goto start
```

Comment:
is it better to put this block in a backoff?

Reply:
backoff is in the `createSocket` call; hence resolving.

Reply:
createSocket already has backoff  built into it.